src/bip32/key_version.rs

```diff
@@ -0,0 +1,109 @@
+use core::{error, fmt};
+
+const VERSION_LEN: usize = 4;
+
+pub enum Version {
+    XPrv,
+    XPub,
+    TPrv,
+    TPub,
+    ZPrv,
+    ZPub,
+    YPrv,
+    YPub,
+}
+
+impl Version {
+    pub fn to_bytes(&self) -> [u8; VERSION_LEN] {
+        fn convert<'a>(version_string: &str, buf: &'a mut [u8; VERSION_LEN]) {
```

Comment:
What about setting directly the correct bytes without doing a conversion so we don't need to perform any operation at runtime?